### PR TITLE
Goodbye self connection

### PR DESCRIPTION
**Pull request by Raynos**
This PR finally gets rid of SelfConnection and SelfPeer

r: @jcorbin @kriskowal

### CODE REVIEW

**Comment by jcorbin** [comment]
:hammer:

File: test/send.js [hunk 1]
```diff
@@ -313,39 +313,53 @@ allocCluster.test('request().send() to self', 1, function t(cluster, assert) {
...
-            resHead: 'head2', resBody: 'msg2',
-            resOk: false,
-            opts: {
-                host: one.hostPort,
-                serviceName: 'one'
+    subOne.waitForIdentified({
+        host: one.hostPort
+    }, onIdentified);
+
+    function onIdentified(err) {
```
Comment (kriskowal) [inline test/send.js]: unfortunate

**Comment by kriskowal** [comment]
:ok: Are you sure about changing the Bad Request strings?

**Comment by Raynos** [comment]
@kriskowal the bad request strings were regressed in a commit that joshua did that broke tcurl.

**Pull request merged**